File: Samples/BasicApplication/BasicApplication/BasicApplication.Script/Imports/FormContexts/Northwind.ProductForm.cs

namespace BasicApplication.Northwind
{
    using Serenity;
    using Serenity.ComponentModel;
    using System;
    using System.Collections;
    using System.Collections.Generic;
    using System.ComponentModel;
    using System.Runtime.CompilerServices;

    public partial class ProductForm : PrefixedContext
    {
        public ProductForm(string idPrefix) : base(idPrefix) {}
    
        public StringEditor ProductName { get { return ById<StringEditor>("ProductName"); } }
        public BooleanEditor Discontinued { get { return ById<BooleanEditor>("Discontinued"); } }
        public LookupEditor SupplierID { get { return ById<LookupEditor>("SupplierID"); } }
        public LookupEditor CategoryID { get { return ById<LookupEditor>("CategoryID"); } }
        public StringEditor QuantityPerUnit { get { return ById<StringEditor>("QuantityPerUnit"); } }
        public DecimalEditor UnitPrice { get { return ById<DecimalEditor>("UnitPrice"); } }
        public StringEditor UnitsInStock { get { return ById<StringEditor>("UnitsInStock"); } }
        public StringEditor UnitsOnOrder { get { return ById<StringEditor>("UnitsOnOrder"); } }
        public StringEditor ReorderLevel { get { return ById<StringEditor>("ReorderLevel"); } }
    }
}

File: common-features/src/demo.northwind/Modules/ServerTypes/Demo/Northwind.ProductForm.ts
import { BooleanEditor, DecimalEditor, ImageUploadEditor, initFormType, IntegerEditor, LookupEditor, PrefixedContext, ServiceLookupEditor, StringEditor } from "@serenity-is/corelib";
import { CategoryDialog } from "../../Category/CategoryDialog";
import { SupplierDialog } from "../../Supplier/SupplierDialog";

export interface ProductForm {
    ProductName: StringEditor;
    ProductImage: ImageUploadEditor;
    Discontinued: BooleanEditor;
    SupplierID: LookupEditor;
    CategoryID: ServiceLookupEditor;
    QuantityPerUnit: StringEditor;
    UnitPrice: DecimalEditor;
    UnitsInStock: IntegerEditor;
    UnitsOnOrder: IntegerEditor;
    ReorderLevel: IntegerEditor;
}

export class ProductForm extends PrefixedContext {
    static readonly formKey = 'Northwind.Product';
    declare private static init: boolean;

    constructor(...args: ConstructorParameters<typeof PrefixedContext>) {
        super(...args);

        if (!ProductForm.init) {
            ProductForm.init = true;

            initFormType(ProductForm, [
                'ProductName', StringEditor,
                'ProductImage', ImageUploadEditor,
                'Discontinued', BooleanEditor,
                'SupplierID', LookupEditor,
                'CategoryID', ServiceLookupEditor,
                'QuantityPerUnit', StringEditor,
                'UnitPrice', DecimalEditor,
                'UnitsInStock', IntegerEditor,
                'UnitsOnOrder', IntegerEditor,
                'ReorderLevel', IntegerEditor
            ]);
        }
    }
}

queueMicrotask(() => [SupplierDialog, CategoryDialog]); // referenced dialogs